vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/TdsMessageDecoder.java

```diff
@@ -41,18 +41,17 @@ protected void decode(ChannelHandlerContext channelHandlerContext, TdsPacket tds
     if (tdsPacket.status() == MessageStatus.END_OF_MESSAGE) {
       if (message == null) {
         message = TdsMessage.newTdsMessageFromSinglePacket(tdsPacket);
-        decodeMessage();
       } else {
         // last packet of this message
         CompositeByteBuf messageData = (CompositeByteBuf) message.content();
         messageData.addComponent(true, tdsPacket.content());
-        decodeMessage();
       }
+      decodeMessage();
     } else {
       if (message == null) {
         // first packet of this message and there will be more packets
         CompositeByteBuf messageData = channelHandlerContext.alloc().compositeDirectBuffer();
-        messageData.addComponent(true, tdsPacket.content());
+        messageData.addComponent(true, tdsPacket.content().slice());
```

Comment:
why add a slice here?

Reply:
It might be a left-over from my branch rebasing. I don't believe it's necessary indeed.

Reply:
I rebased the PR and removed this